src/Tickets/Commerce/Utils/Value.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Utils;
+
+use Tribe\Values\Abstract_Currency;
+
+class Value extends Abstract_Currency  {
+
+	private $currency_code = 'USD';
+
+	private $currency_separator_decimal = '.';
+
+	private $currency_separator_thousands = ',';
+
+	private $currency_symbol = '$';
+
+	private $currency_symbol_position = 'prefix';
+
+	/**
+	 * @inheritDoc
+	 */
+	public function set_up_currency_details() {
+		$this->currency_code = Currency::get_currency_code();
+		$this->currency_symbol = Currency::get_currency_symbol( $this->get_currency_code() );
+		$this->currency_symbol_position = Currency::get_currency_symbol_position( $this->get_currency_code() );
+		$this->currency_separator_decimal = Currency::get_currency_separator_decimal( $this->get_currency_code() );
+		$this->currency_separator_thousands = Currency::get_currency_separator_thousands( $this->get_currency_code() );
+	}
+
+	public static function build_list( $values ) {
+		return array_map( function( $value ) {
+
+			if ( \is_a(  $value,'Tribe\Values\Abstract_Currency' ) ) {
+				return $value;
+			}
+
+			return new self( $value );
+		}, $values );
+	}
```

Comment:
I am confused into what this method is used for.

Reply:
This method takes a list of numeric values and returns a list of Value objects instantiated from those values. This is a helper to facilitate limiting the scope of methods such as `Value->total()` that take a list of objects, instead of a list of primitives.